CMakeLists.txt

```diff
@@ -4,22 +4,31 @@ cmake_policy(SET CMP0048 NEW)
 
 project(cpp_lmdb VERSION 0.1)
 
+set (CMAKE_CXX_STANDARD 23)
+
 set(3P_DIR ${CMAKE_CURRENT_SOURCE_DIR}/3p)
 
-add_subdirectory(${3P_DIR}/googletest)
 add_subdirectory(${3P_DIR}/lmdb_cmake)
 
+if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
+    add_compile_options(-Wall -Wextra -fconcepts-diagnostics-depth=5)
+elseif("${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang")
+    add_compile_options(-Wall -Wextra)
+    add_compile_options(-std=c++2b -stdlib=libc++)
```

Comment:
GUN clang-17 supports c++23 but unfortunately AppleClang 15 does not, only c++2b. :(